harness/determined/launch/autodeepspeed.py

```diff
@@ -237,9 +196,36 @@ def main(train_entrypoint: str) -> int:
 
     full_cmd = pid_server_cmd + cmd + pid_client_cmd + log_redirect_cmd + harness_cmd
 
-    exit_code = subprocess.Popen(full_cmd).wait()
+    # Chief worker also needs to run sshd when using pdsh and multi-machine training.
+    multi_machine = len(info.container_addrs) > 1
+    if multi_machine:
+        # Create the environment file that will be passed by deepspeed to individual ranks.
+        create_deepspeed_env_file()
+        # Set custom PDSH args:
+        # * bypass strict host checking
+        # * -p our custom port
+        # * other args are default ssh args for pdsh
+        os.environ["PDSH_SSH_ARGS"] = (
+            "-o PasswordAuthentication=no -o StrictHostKeyChecking=no "
+            f"-p {constants.DTRAIN_SSH_PORT} -2 -a -x %h"
+        )
+
+        sshd_process = subprocess.Popen(run_sshd_command)
+
+        try:
+            # Chief machine waits for every worker's sshd to be available.  All machines should be
+            # close to in-step by now because all machines just finished synchronizing rendezvous
+            # info.
+            for peer_addr in info.container_addrs:
+                util.check_sshd(peer_addr, time.time() + 5, constants.DTRAIN_SSH_PORT)
+
+            return_code = subprocess.Popen(full_cmd).wait()
+        finally:
+            sshd_process.kill()
+            sshd_process.wait()
+        return return_code
 
-    return exit_code
+    return subprocess.Popen(full_cmd).wait()
```

Comment:
very small nitpick:

since this line is so much smaller than the above function, can you invert the if logic?

```python
if not multi_machine:
    return subprocess.Popen(full_cmd).wait()
...
# Create the environment file that will be passed by deepspeed to individual ranks.
create_deepspeed_env_file()
```